x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -3548,6 +3559,34 @@ export class RulesClient {
         })
       );
     }
+
+    // check for actions using frequency params if the rule has global frequency params defined
+    if (usesGlobalFreqParams) {
+      const actionsWithFrequency = actions.filter((action) => Boolean(action.frequency));
+      if (actionsWithFrequency.length) {
+        throw Boom.badRequest(
+          i18n.translate('xpack.alerting.rulesClient.validateActions.mixAndMatchFreqParams', {
+            defaultMessage:
+              'Cannot mix and match per-action frequency params when notify_when or throttle are globally defined: {groups}',
+            values: {
+              groups: actionsWithFrequency.map((a) => a.group).join(', '),
+            },
+          })
+        );
+      }
+    } else {
+      const actionsWithoutFrequency = actions.filter((action) => !action.frequency);
+      if (actionsWithoutFrequency.length) {
```

Comment:
That's valid but how can this happen?

Reply:
This checks to make sure that every action has a `frequency` parameter.